### DISCUSSION TITLE
Intent to remove: IE11 support

### DISCUSSION THREAD

**Discussion by triskweline** [unpoly/unpoly#340] [Announcements]
Unpoly traditionally went to great lengths to support old versions of Internet Explorer. This added a lot of unwanted code and complexity to the codebase.

On June 15th 2022 [Microsoft will finally bury IE11](https://techcommunity.microsoft.com/t5/windows-it-pro-blog/internet-explorer-11-desktop-app-retirement-faq/ba-p/2366549) for the vast majority of its remaining users. After this date, the IE11 desktop application will be disabled through a Windows 10 cumulative monthly update and will redirect to Microsoft Edge if a user tries to access it.

I'm planning to remove any IE11 specific code from Unpoly in the first half of 2022. This will result in smaller bundle sizes and faster execution times for the vast majority of users.

Functions that have a native counterpart in modern versions (like [`up.element.closest()`](https://unpoly.com/up.element.closest)) will be removed, but remain available through [`unpoly-migrate.js`](https://unpoly.com/changes/upgrading).

There is probably going to be one more release with full IE11 support.

**Comment by johndwells** [comment]
🎉 IE11 is now a thing of the past! I can't wait to see this intent actioned. :)

**Comment by triskweline** [comment]
It's already gone in `master` :)
